src/test/java/api/loans/OverrideCheckOutByBarcodeTests.java

```diff
@@ -0,0 +1,267 @@
+package api.loans;
+
+import static api.support.builders.ItemBuilder.CHECKED_OUT;
+import static api.support.matchers.ItemStatusCodeMatcher.hasItemStatus;
+import static api.support.matchers.TextDateTimeMatcher.isEquivalentTo;
+import static api.support.matchers.ValidationErrorMatchers.hasErrorWith;
+import static api.support.matchers.ValidationErrorMatchers.hasMessage;
+import static api.support.matchers.ValidationErrorMatchers.hasParameter;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.net.MalformedURLException;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.folio.circulation.support.http.client.Response;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Test;
+
+import api.support.APITests;
+import api.support.builders.LoanPolicyBuilder;
+import api.support.builders.OverrideCheckOutByBarcodeRequestBuilder;
+import api.support.matchers.UUIDMatcher;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class OverrideCheckOutByBarcodeTests extends APITests {
+
+  private static final DateTime TEST_LOAN_DATE =
+    new DateTime(2019, 4, 10, 11, 35, 48, DateTimeZone.UTC);
+  private static final DateTime TEST_DUE_DATE =
+    new DateTime(2019, 4, 20, 11, 30, 0, DateTimeZone.UTC);
+  private static final String TEST_COMMENT = "Some comment";
+
+  @Test
+  public void canOverrideCheckoutWhenItemIsNotLoanable()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    IndividualResource steve = usersFixture.steve();
+    final UUID checkoutServicePointId = UUID.randomUUID();
+
+    setNotLoanablePolicy();
+    IndividualResource response = loansFixture.overrideCheckOutByBarcode(
+      new OverrideCheckOutByBarcodeRequestBuilder()
+        .forItem(smallAngryPlanet)
+        .to(steve)
+        .at(checkoutServicePointId)
+        .on(TEST_LOAN_DATE)
+        .withDueDate(TEST_DUE_DATE)
+        .withComment(TEST_COMMENT)
+    );
+
+    final JsonObject loan = response.getJson();
+
+    assertThat("action should be checkedoutThroughOverride",
+      loan.getString("action"), is("checkedoutThroughOverride"));
```

Comment:
The expected action in the UI is `Checked out through override`.

I'm aware actions in the past have not been this text specifically. Is there a reason why we are using the compressed version here?

If we are going to continue to use the compressed form, should it be `checkedOutThroughOverride`?

Reply:
This keys are mapped on UI: https://github.com/folio-org/ui-users/blob/979f5e93a3680231dbf74e7e662a68a99f93736a/src/data/loanActionMap.js
Changed `checkedoutThroughOverride` to `checkedOutThroughOverride`
and  `Renewed through override` to 'renewedThroughOverride'.
I'm going to provide a link to corresponding pull request on UI. Do we need to change interface version for this case?

Reply:
This isn't an interface change as none of this is described in the interface. This is an example of a hidden implementation to implementation dependency. 

The implementation version is already at a new major version, so no further change is needed.

Does that make sense?

Reply:
Yes, makes sense to me, thanks.